build.zig

```diff
@@ -9,58 +9,76 @@ const Mode = builtin.Mode;
 
 pub fn build(b: *Builder) !void {
     const target = Target{
-        .Cross = std.build.CrossTarget{
+        .Cross = Target.Cross{
             .arch = .i386,
             .os = .freestanding,
             .abi = .gnu,
         },
     };
 
+    const test_target = Target{
+        .Cross = Target.Cross{
+            .arch = .i386,
+            .os = .linux,
+            .abi = .gnu,
+        },
+    };
+
     const target_str = switch (target.getArch()) {
         .i386 => "x86",
         else => unreachable,
     };
 
+    const main_src = "src/kernel/kmain.zig";
+    const constants_path = try fs.path.join(b.allocator, &[_][]const u8{ "src/kernel/arch", target_str, "constants.zig" });
+
     const build_mode = b.standardReleaseOptions();
     const rt_test = b.option(bool, "rt-test", "enable/disable runtime testing") orelse false;
 
-    const main_src = "src/kernel/kmain.zig";
-
     const exec = b.addExecutable("pluto", main_src);
-    const constants_path = try fs.path.join(b.allocator, &[_][]const u8{ "src/kernel/arch", target_str, "constants.zig" });
     exec.addPackagePath("constants", constants_path);
-    exec.setBuildMode(build_mode);
+    exec.setOutputDir("zig-cache");
```

Comment:
Do we need to set this explicitly? zig-cache is the default anyway and I'm not sure how this will interfere with the use specifying a custom output dir when running `zig build`.

Reply:
I need to set is explicitly bc I use `getOutputPath()` and this is only set when `make` is called at the end.

Reply:
I see, that's a weird restriction. The only problem is that it outputs to zig-cache even if the user specifies --cache-dir when building. It's fine if there isn't a workaround but it would be nice if we didn't have to set things to something hardcoded.

Reply:
If you specify `--cache-dir` it overrides the cache directory even if one is hardcoded.

Reply:
It does but it still puts the pluto binary in zig-cache. Could we set the output dir to whatever cache dir has been set?

Reply:
Ah yes, it does.